src/viam_mlmodelservice_triton_impl.cpp

```diff
@@ -419,18 +444,55 @@ class Service : public vsdk::MLModelService, public vsdk::Stoppable, public vsdk
         }
         state->model_name = std::move(*model_name_string);
 
-        auto model_version = attributes->find("model_version");
-        if (model_version != attributes->end()) {
-            auto* const model_version_value = model_version->second->get<double>();
-            if (!model_version_value || (*model_version_value < 1) ||
-                (std::nearbyint(*model_version_value) != *model_version_value)) {
+        // Pull the model repository path out of the configuration.
+        auto model_repo_path = attributes->find("model_repository_path");
+        if (model_repo_path == attributes->end()) {
+            // With no model repository path, we try to construct our own by symlinking a single
+            // model path.
+            symlink_mlmodel_(*state.get());
+            state->model_repo_path = std::move(std::getenv("VIAM_MODULE_DATA"));
+            state->model_version = 1;
+        } else {
+            auto* const model_repo_path_string = model_repo_path->second->get<std::string>();
+            if (!model_repo_path_string || model_repo_path_string->empty()) {
+                std::ostringstream buffer;
+                buffer << service_name
+                       << ": Non-empty string parameter `model_repository_path` is either not "
+                          "a string or is an empty string";
+                throw std::invalid_argument(buffer.str());
+            }
+            state->model_repo_path = std::move(*model_repo_path_string);
+
+            // If you specify your own model repo path, you need to specify your own model version
```

Comment:
nit: s/need to/may, since if you don't set it you get `-1` which means latest per earlier discussion.